assets/src/hooks/DagreHook.ts

```diff
@@ -21,14 +21,17 @@ const DagreHook = {
     const graph = createDagre(this.el, width, height);
     this.graph = graph;
 
+    let hasInteracted = false;
+
     window.onresize = () => {
       if (!graph || graph.get("destroyed")) return;
       if (!this.el || !this.el.scrollWidth || !this.el.scrollHeight) return;
       this.graph.changeSize(this.el.scrollWidth, this.el.scrollHeight);
     };
 
     document.getElementById("dagre-relayout")?.addEventListener("click", () => {
-      this.graph.updateLayout({ sortByCombo: true });
+      this.graph.updateLayout(defaultLayout);
+      this.graph.destroyLayout();
```

Comment:
How does updating layout and then destroying it work?

Reply:
Okay, so we can update internal data of a graph with two functions: `graph.data` or `graph.changeData`. The first simply replaces existing data under the hood (but doesn't render it), when the latter diffs it with new data, updates existing items and **updates layout**.
If there are no new nodes in incoming data and user has interacted with the graph, we want to use `graph.changeData`, but we can't allow the layout to update, since it moves the elements.
Hence, we need to keep the layout algorithm off when we know the user has been modifying the graph.

Reply:
Added conditional check for whether destroying layout is necessary in 214e805, maybe will be more descriptive too.